.azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml

```diff
@@ -165,30 +165,37 @@ jobs:
                 . (Join-Path '$(moduleRepoRoot)' 'utilities' 'pipelines' 'tokensReplacement' 'Convert-TokensInParameterFile.ps1')
 
                 # Load Settings File
-                $Settings = Get-Content -Path (Join-Path '$(moduleRepoRoot)' 'settings.json') | ConvertFrom-Json
-
-                # Initialize Default Parameter File Tokens
-                $DefaultParameterFileTokens = @(
-                    @{ Name = 'resourceGroupName'; Value = '${{ parameters.resourceGroupName }}' }
-                    @{ Name = 'subscriptionId';    Value = '${{ parameters.subscriptionId }}' }
-                    @{ Name = 'managementGroupId'; Value = '${{ parameters.managementGroupId }}' }
-                    @{ Name = "tenantId";          Value = '$(ARM_TENANT_ID)' }
-                    @{ Name = "deploymentSpId";    Value = '$(DEPLOYMENT_SP_ID)' }
-                ) | ForEach-Object { [PSCustomObject]$PSItem }
-
-                # Get additional Custom Parameter File Tokens from input
-                Write-Verbose 'Additional Custom Parameter File Tokens: ${{ deploymentBlock.customParameterFileTokens }}' -Verbose
-                $OtherCustomParameterFileTokens = '${{  deploymentBlock.customParameterFileTokens }}' | ConvertFrom-Json
+                $Settings = Get-Content -Path (Join-Path '$(moduleRepoRoot)' 'settings.json') | ConvertFrom-Json -AsHashTable
 
                 # Construct Token Function Input
                 $ConvertTokensInputs = @{
-                    ParameterFilePath                 = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
-                    DefaultParameterFileTokens        = $DefaultParameterFileTokens
-                    OtherCustomParameterFileTokens    = $OtherCustomParameterFileTokens
-                    LocalCustomParameterFileTokens    = $Settings.parameterFileTokens.localTokens.tokens
-                    TokenPrefix                       = $Settings.parameterFileTokens.tokenPrefix
-                    TokenSuffix                       = $Settings.parameterFileTokens.tokenSuffix
+                    Tokens = @{
```

Comment:
I think the 'default' tokens should still live outside this as a separate object, similar to the 'TokenMap'. This makes it easy to distinguish and modify. The `Tokens` property can just be appended to process more token types. Maybe something like this:

```pwsh
                $ConvertTokensInputs = @{
                    ParameterFilePath                 = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
                    DefaultParameterFileTokens        = $DefaultParameterFileTokens
                    OtherCustomParameterFileTokens    = $OtherCustomParameterFileTokens
                    LocalCustomParameterFileTokens    = $Settings.parameterFileTokens.localTokens.tokens
                    TokenPrefix                       = $Settings.parameterFileTokens.tokenPrefix
                    TokenSuffix                       = $Settings.parameterFileTokens.tokenSuffix
                    Tokens = @{}
                    FilePath   = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
               }
```
Then something like this for default tokens:

```pwsh
$ConvertTokensInputs.Tokens += $DefaultTokens
```

Reply:
I disagree as it just adds unnecessary lines of code, but then again I don't have a strong opinion about it, so will do if that's how you prefer it

Reply:
done